### ISSUE TITLE
[Feature Request] Add more control to histograms

### ISSUE THREAD

**Issue by TatriX** [38/plotters#75]
## What is the feature ?

Ability to control color of individual bars in the histogram.

## Why do you need the feature / What is the use case ?
Take a look at this plot:
![2019-10-30-154312_1228x626_scrot](https://user-images.githubusercontent.com/703150/67868303-13c01900-fb2c-11e9-923d-31a636969480.png)
It shows daily data. I'd like to color 6th and 7th bars with different color, so it would be easy to see weekends.

Then it would be also nice to have a better control over ticks placement. For instance I'd like to put ticks every 7 bars to highlight full weeks.

**Comment by 38** [comment]
I don't know if you realized this, series is just a iterator of elements. Basically bar chart can be done with 
```rust
     ctx.draw_series( 
             data.map(|(bucket, value)| {
                    let rect = Rectangle::new([(bucket - 1, 0), (bucket, value)], BLUE.filled());
                    rect.margin(0, 0, 5, 5); 
                    rect
            })
     );
```

And of course you can control the color. I am not very satisfy with current implementation of histogram, and I am thinking about some better solution as well. If you have any thoughts, please let me know :)

For your second question, this also can be done by implementing a wrapper for ranged value and you can control what key point you want to emit. The key trait to implement is called `Ranged`, see docs at https://docs.rs/plotters/0.2.11/plotters/coord/trait.Ranged.html. 
 
The logarithmic coordinate is supported in this way, please see code at https://github.com/38/plotters/blob/master/src/coord/logarithmic.rs. 

 And of course both suggest make sense to me and please let me think how to make built-in support for these.

**Comment by TatriX** [comment]
It's not yet clear to me what can passed to `draw_series`. Currently my brain stops at `E: Drawable<DB>` from [here](https://github.com/38/plotters/blob/458be85f7bba0e708fbf95c242ec237dab6720dd/src/chart/context.rs#L216).
I guess passing some kind of data structure should do the trick:
```rust
draw_series(LineSeries::new(data.map(|x, y| Point{x, y, color: Some(rand_color()) }), &RED))
```
So color defaults to `RED` but can be overriden by individual data point.

**Comment by 38** [comment]
Hmm, an element is actually combination of two trait, one is `PointCollection` which holds the vertices another one is `Drawable<DB>` means once the coordinate mapping is done, it can be drawn on to the specific backend. 

And anything under `element` module should be ready to pass into `draw_series`. 

The element system is a little bit undocumented unfortunately, the only doc I have is https://docs.rs/plotters/0.2.11/plotters/element/index.html

**Comment by 38** [comment]
A quick update on this. I changed the histogram API which allows you to passing a closure that retutrns style based on the values. 

Check here for details, let me know if this works for you:)

https://github.com/38/plotters/blob/8e99cdd69b9708f571c46d07161048dbcfe344a5/src/series/histogram.rs#L59

**Comment by 38** [comment]
Also, now you can use code like
```rust
.build_ranged((0..100).group_by(7), 10..20)?
```
to make the key point a multiply of 7

**Comment by TatriX** [comment]
This looks amazing! Will try soon.
BTW, do you have any plans on making range being set in an automatic fashion? 
Currently I need to collect data to for example into Vec, then find min/max values and only then I can draw series.

**Comment by TatriX** [comment]
Yep, both features work like a charm, thanks!
